docs/generated/reference/output/reference/api-docs/index.html

```diff
@@ -85,11 +85,11 @@ <h3 id="certificatespec-v1alpha1">CertificateSpec v1alpha1</h3>
 </tr>
 <tr>
 <td><code>commonName</code><br /> <em>string</em></td>
-<td>CommonName is a common name to be used on the Certificate</td>
+<td>CommonName is a common name to be used on the Certificate. If no CommonName is given, then the first entry in DNSNames is used as the CommonName. The CommonName should have a length shorter than 64 bytes to avoid generating invalid CSRs; in order to have longer domain names, set the CommonName (or first DNSNames entry) to have less than 64 bytes, and then add the longer domain name to DNSNames.</td>
```

Comment:
I think for both the `commonName` and `dnsNames` description it might be worth emphasizing that this "promote the first SAN to CN" behaviour isn't standardized in RFC 8555 and just a typical implementation choice. It can't be relied on absolutely (unfortunately).

Reply:
what is the recommended CN, so if it can't be > 64 bytes, wouldn't the following breaking it as dnsnames could be very long > 64 bytes?
"If you don't specify a commonName, it will be set to dnsNames[0]. If you do
specify commonName it will become dnsNames[0] (and all existing dnsNames
are shifted by 1, and then deduped in case it already existed in dnsNames)."

Reply:
That's up to the server. Historically Boulder/Let's Encrypt would reject the order if the first DNS type SAN was > 64 bytes.

Reply:
In our case (cert-manager), we actually *do* currently enforce this, in order to make the comparison logic to check if a certificate is up to date 'simpler'.

In future I think we're going to revisit this logic (as we discussed in another thread), which means we could potentially relax this requirement.

What is the reasoning for boulder setting it at all if it is a deprecated field? For compatibility with older clients? Given how long its been deprecated for it seems surprising that it is mandated that boulder sets it 🤔 (but I am aware that the x509 gods move in mysterious ways and I'm sure there's some crazy reason why you've done this 🙈)

Reply:
@munnerz Yeah, this gets square into the muck of the web PKI. My tl;dr understanding of the current situation is that we can't omit the CN entirely because it breaks a non-trivial amount of relying party software. We planned to work around that by putting something other than a subject domain into the CN (e.g. the cert serial number) but this ends up being forbidden by the CABF baseline reqs. in their current state. There was some ballot work to try and fix this but it stalled and nobody has unstuck it. I think there are a few community forum threads that summarize this in more detail (i'm OOO today and don't have a link handy).